analytics_data_api/utils.py

```diff
@@ -84,3 +94,92 @@ def date_range(start_date, end_date, delta=datetime.timedelta(days=1)):
     while cur_date < end_date:
         yield cur_date
         cur_date += delta
+
+
+def get_course_report_download_details(course_id, report_name):
+    """
+    Determine the path that the report file should be located at,
+    then return metadata sufficient for downloading it.
+    """
+    report_location_template = getattr(settings, 'COURSE_REPORT_FILE_LOCATION_TEMPLATE')
+    report_location = report_location_template.format(
+        course_id=course_id,
+        report_name=report_name
+    )
+    if getattr(settings, 'DEFAULT_FILE_STORAGE', None) != AWS_S3_FILE_STORAGE_CLASS:
+        # At present, we do not support linking from storage other than S3.
+        raise CannotCreateReportDownloadLinkError
+
+    if not default_storage.exists(report_location):
+        raise ReportFileNotFoundError(course_id=course_id, report_name=report_name)
+
+    try:
+        last_modified = default_storage.modified_time(report_location)
+    except (NotImplementedError, AttributeError):
+        last_modified = None
+
+    try:
+        download_size = default_storage.size(report_location)
+    except (NotImplementedError, AttributeError):
+        download_size = None
+
+    # Note that course IDs contain characters that aren't valid for filenames in
+    # Windows. However, we'll trust the client to save them with the appropriate
+    # alternate naming scheme (replace ':' with '_' and so on)
+    download_filename = '{}-{}-{}.csv'.format(
+        course_id,
+        report_name,
+        # We need a date for the filename; if we don't know when it was last modified,
+        # use the current date and time to stamp the filename.
+        (last_modified or datetime.datetime.utcnow()).strftime('%Y%m%dT%H%M%S')
```

Comment:
Nit: could you append a 'Z' to the datetime format?  It's not ISO-8601 or anything, but at least gives a clue that the datetime is in UTC.